src/pages/WalletProviders.tsx

```diff
@@ -0,0 +1,206 @@
+import { useRedux } from "hooks/useRedux";
+import { useState, useEffect } from "react";
+import { AppDispatch } from "store";
+import { getWalletsAction, updateWalletAction } from "store/ducks/wallets";
+
+export const WalletProviders = () => {
+  const [updateWalletModal, setUpdateWalletModal] = useState({
+    visible: false,
+    walletId: "",
+    enabled: false,
+  });
+
+  const { wallets } = useRedux("wallets");
+  const dispatch: AppDispatch = useDispatch();
+
+  useEffect(() => {
+    if (!wallets.status) {
+      dispatch(getWalletsAction());
+    }
+  }, [wallets.status, dispatch]);
+
+  const myWallets = wallets?.items.filter((e) => e.enabled);
+  const avalaibleWallets = wallets?.items.filter((e) => !e.enabled);
+
+  const handleCloseModal = () => {
+    setUpdateWalletModal({ visible: false, walletId: "", enabled: false });
+  };
+
+  const handleUpdateWallet = async (walletId: string, enabled: boolean) => {
+    await dispatch(updateWalletAction({ walletId, enabled }));
+    window.location.reload();
+  };
+
+  const renderWalletCard = (
+    walletName: string,
+    walletId: string,
+    link: string,
+    enabled: boolean,
+    assets: string[],
+  ) => {
+    if (wallets.errorString) {
+      return (
+        <Notification variant="error" title="Error">
+          {wallets.errorString}
+        </Notification>
+      );
+    }
+
+    return (
+      <>
+        <Card noPadding>
+          <div className="WalletCard">
+            <div className="WalletCard__title">
+              <div className="WalletCard__item">
+                <div>
+                  <div className="WalletCard__item">
+                    <Title size="lg">{walletName}</Title>
+
+                    <Icon.ExternalLink
+                      className="ExternalLinkIcon"
+                      onClick={() => window.open(link, "_blank")}
+                    />
+                  </div>
+                </div>
+
+                <Toggle
+                  id={walletId}
+                  checked={enabled}
+                  onChange={() => {
+                    setUpdateWalletModal({
+                      visible: true,
+                      walletId: walletId,
+                      enabled: enabled,
+                    });
+                  }}
+                />
+              </div>
+            </div>
+
+            <div className="WalletCard__walletData">
+              <div className="WalletCard--flexCols">
+                <label className="WalletCard__item__label">
+                  <Icon.Assets classname="WalletProvidersIcon" />
+                  Supported assets
+                </label>
+                <div className="WalletCard__item__value">
+                  {assets?.join(", ")}
+                </div>
+              </div>
+            </div>
+          </div>
+        </Card>
+
+        {/* Enable/Disable wallet modal */}
+        <Modal visible={updateWalletModal.visible} onClose={handleCloseModal}>
```

Comment:
The modal shouldn't be in the `map`. We need to have only one modal on this page. It's rendering a modal for every wallet provider now.

![image](https://github.com/stellar/stellar-disbursement-platform-frontend/assets/7346473/da803757-49ed-4d9a-a43f-6d2511ef17e9)